packages/@sanity/form-builder/src/inputs/SelectInput.js

```diff
@@ -45,19 +50,19 @@ export default class StringSelect extends React.Component<Props> {
 
   render() {
     const {value, readOnly, markers, type, level, ...rest} = this.props
-
     const items = toSelectItems(type.options.list || [])
 
     const currentItem = items.find(item => item.value === value)
 
     const isRadio = type.options && type.options.layout === 'radio'
+
     return (
       <FormField markers={markers} level={level} label={type.title} description={type.description}>
         {isRadio ? (
           // todo: make separate inputs
           <RadioSelect
             {...rest}
-            name={type.name}
+            name={this.name || type.name}
```

Comment:
So `this.name` will never be false, right? Can just remove `|| type.name` from here I think.